apps/api/src/modules/webhook/webhook.service.ts

```diff
@@ -0,0 +1,71 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { Webhook } from './entities/webhook.entity';
+import { Notification } from '../notifications/entities/notification.entity';
+import axios from 'axios';
+
+@Injectable()
+export class WebhookService {
+  protected readonly logger = new Logger(WebhookService.name);
+
+  constructor(
+    @InjectRepository(Webhook)
+    private readonly webhookRepository: Repository<Webhook>,
+  ) {}
+
+  async registerWebhook(providerId: number, webhookUrl: string): Promise<void> {
+    const webhook = new Webhook();
+    webhook.providerId = providerId;
+    webhook.webhookUrl = webhookUrl;
+
+    await this.webhookRepository.save(webhook);
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  async triggerWebhook(notification: Notification): Promise<void> {
+    const maxRetries = 5;
+    let attempts = 0;
+
+    this.logger.log(
+      `Triggering webhook for notification with providerId: ${notification.providerId}`,
+    );
+
+    while (attempts < maxRetries) {
+      try {
+        const webhook = await this.webhookRepository.findOneBy({
```

Comment:
This should add check for status as ACTIVE too.

Reply:
done